src/Libraries/CoreNodeModels/Equals.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dynamo.Graph.Nodes;
+using Newtonsoft.Json;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.DSASM;
+using ProtoCore.Utils;
+
+namespace CoreNodeModels
+{
+    [NodeName("==")]
+    [NodeDescription("Equals with tolerance input")]
+    [NodeCategory("Core.Math")]
+    [NodeSearchTags("Equals")]
+    [InPortTypes("double ", "double", "double")]
+    [OutPortTypes("bool")]
+    [IsDesignScriptCompatible]
+    public class Equals : NodeModel
+    {
+        private static readonly DoubleNode tolerancePortDefaultValue = new DoubleNode(MathUtils.Tolerance);
+
+        [JsonConstructor]
+        private Equals(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) : base(inPorts, outPorts)
+        {
+            ArgumentLacing = LacingStrategy.Auto;
+            inPorts.ElementAt(2).DefaultValue = tolerancePortDefaultValue;
+        }
+
+        public Equals()
+        {
+            ArgumentLacing = LacingStrategy.Auto;
+
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("lhs", "integer or double value")));
```

Comment:
can we use x and y instead of lhs and rhs?

Reply:
Done